CMake/functions/devilutionx_library.cmake

```diff
@@ -37,7 +37,13 @@ function(add_devilutionx_library NAME)
     target_link_libraries(${NAME} PUBLIC $<${DEVILUTIONX_STATIC_CXX_STDLIB_GENEX}:-static-libgcc;-static-libstdc++>)
   endif()
 
-  if(NOT CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+  if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+    if(CMAKE_BUILD_TYPE STREQUAL "Debug")

```

Comment:
I think you may need to use generator expressions for Visual Studio, e.g.:

```cmake
set(_is_msvc "$<OR:$<CXX_COMPILER_ID:MSVC>,$<C_COMPILER_ID:MSVC>>")
target_compile_options(${NAME} PUBLIC $<$<AND:${_is_msvc},$<CONFIG:Debug>>:/ZI>)
target_link_options(${NAME} PUBLIC $<$<AND:${_is_msvc},$<CONFIG:Debug>>:/INCREMENTAL>)
```

Reply:
I moved it to the global cmake file and tried using [CMAKE_MSVC_DEBUG_INFORMATION_FORMAT])(https://cmake.org/cmake/help/latest/variable/CMAKE_MSVC_DEBUG_INFORMATION_FORMAT.html).
This avoids the warning that both /Zi and /ZI are set.
This is only available since 3.25, so I only enabled Edit & Continue in a newer CMake version.

I'm open to other approaches. I'm not a CMake expert.